src/internal/_eqDeep.js

```diff
@@ -0,0 +1,45 @@
+var eq = require('../eq');
+var keys = require('../keys');
+var type = require('../type');
+
+// The algorithm used to handle cyclic structures
+// is heavily inspired by underscore.js'
+// implementation
+module.exports = function _eqDeep(a, b, stackA, stackB) {
+    for (var idx = stackA.length - 1; idx >= 0; idx--) {
+        if (stackA[idx] === a) {
+            return stackB[idx] === b;
+        }
+    }
+    stackA.push(a);
+    stackB.push(b);
+    var result;
+    switch (type(a)) {
+    case 'Array':
+    case 'Object':
+        if (keys(a).length !== keys(b).length ||
+            type(a) !== type(b)) {
+            result = false;
+        } else {
+            var keysA = keys(a);
+            var length = keysA.length - 1;
+            result = true;
+
+            while (result && length >= 0) {
```

Comment:
Again here, we usually loop from 0 up (although this is much less of an issue I think than the no-for-loop one.)